packages/dotnet-sdk/src/TeamsFx/TeamsUserCredential2.cs

```diff
@@ -0,0 +1,531 @@
+        /// </code>
+        /// </example>
+        /// <remarks>
+        /// Can only be used within Teams.
+        /// </remarks>
+        public async override ValueTask<Azure.Core.AccessToken> GetTokenAsync(TokenRequestContext requestContext, CancellationToken cancellationToken)
+        {
+            var scopes = requestContext.Scopes;
+            var ssoToken = await GetSsoTokenAsync().ConfigureAwait(false);
+            if (scopes == null || scopes.Length == 0)
+            {
+                _logger.LogInformation("Get SSO token");
+                return ssoToken.ToAzureAccessToken();
+            }
+            else
+            {
+                var scopeString = string.Join(' ', scopes);
+                _logger.LogInformation($"Get access token with scopes: {scopeString}");
+                var cacheKey = Utils.GetCacheKey(ssoToken.Token, scopeString, _authenticationOptions.ClientId);
+                var cachedToken = await GetTokenFromCacheAsync(cacheKey).ConfigureAwait(false);
+
+                if (cachedToken != null)
+                {
+                    if (!cachedToken.NearExpiration())
+                    {
+                        _logger.LogTrace("Get access token from cache");
+                        return cachedToken.ToAzureAccessToken();
+                    }
+                    else
+                    {
+                        _logger.LogTrace("Cached access token is expired");
+                    }
+                }
+                else
+                {
+                    _logger.LogTrace("No cached access token");
+                }
+
+                var accessToken = await GetAndCacheAccessTokenFromSimpleAuthServer(scopeString).ConfigureAwait(false);
+                return accessToken.ToAzureAccessToken();
+            }
+        }
+
+        /// <summary>
+        /// Sync version is not supported now. Please use GetTokenAsync instead.
+        /// </summary>
+        public override Azure.Core.AccessToken GetToken(TokenRequestContext requestContext, CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Dispose.
+        /// </summary>
+        /// <returns></returns>
+        public async ValueTask DisposeAsync()
+        {
+            if (_teamsSdkTask.IsValueCreated)
+            {
+                var instance = await _teamsSdkTask.Value.ConfigureAwait(false);
+                await instance.DisposeAsync().ConfigureAwait(false);
+            }
+        }
+
+        private async ValueTask<IJSObjectReference> ImportTeamsSdk(IJSRuntime jsRuntime)
+        {
+            try
+            {
+                await jsRuntime.InvokeVoidAsync("import", "https://statics.teams.cdn.office.net/sdk/v1.7.0/js/MicrosoftTeams.min.js").ConfigureAwait(false);
```

Comment:
VS scaffolded project use v1.10.0 Teams library, maybe we can use the latest version?

Reply:
Fixed.